bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/HueGroupHandler.java

```diff
@@ -0,0 +1,373 @@
+            if (tmpBrightness != null) {
+                lastSentBrightness = tmpBrightness;
+            }
+            Integer tmpColorTemp = groupState.getColorTemperature();
+            if (tmpColorTemp != null) {
+                lastSentColorTemp = tmpColorTemp;
+            }
+            bridgeHandler.updateGroupState(group, groupState);
+        } else {
+            logger.warn("Command sent to an unknown channel id: {}:{}", getThing().getUID(), channel);
+        }
+    }
+
+    private @Nullable StateUpdate convertColorTempChangeToStateUpdate(IncreaseDecreaseType command, FullGroup group) {
+        StateUpdate stateUpdate = null;
+        Integer currentColorTemp = getCurrentColorTemp(group.getState());
+        if (currentColorTemp != null) {
+            int newColorTemp = LightStateConverter.toAdjustedColorTemp(command, currentColorTemp);
+            stateUpdate = new StateUpdate().setColorTemperature(newColorTemp);
+        }
+        return stateUpdate;
+    }
+
+    private @Nullable Integer getCurrentColorTemp(@Nullable State groupState) {
+        Integer colorTemp = lastSentColorTemp;
+        if (colorTemp == null && groupState != null) {
+            colorTemp = groupState.getColorTemperature();
+        }
+        return colorTemp;
+    }
+
+    private @Nullable StateUpdate convertBrightnessChangeToStateUpdate(IncreaseDecreaseType command, FullGroup group) {
+        StateUpdate stateUpdate = null;
+        Integer currentBrightness = getCurrentBrightness(group.getState());
+        if (currentBrightness != null) {
+            int newBrightness = LightStateConverter.toAdjustedBrightness(command, currentBrightness);
+            stateUpdate = createBrightnessStateUpdate(currentBrightness, newBrightness);
+        }
+        return stateUpdate;
+    }
+
+    private @Nullable Integer getCurrentBrightness(@Nullable State groupState) {
+        Integer brightness = lastSentBrightness;
+        if (brightness == null && groupState != null) {
+            if (!groupState.isOn()) {
+                brightness = 0;
+            } else {
+                brightness = groupState.getBrightness();
+            }
+        }
+        return brightness;
+    }
+
+    private StateUpdate createBrightnessStateUpdate(int currentBrightness, int newBrightness) {
+        StateUpdate lightUpdate = new StateUpdate();
+        if (newBrightness == 0) {
+            lightUpdate.turnOff();
+        } else {
+            lightUpdate.setBrightness(newBrightness);
+            if (currentBrightness == 0) {
+                lightUpdate.turnOn();
+            }
+        }
+        return lightUpdate;
+    }
+
+    @Override
+    public void channelLinked(ChannelUID channelUID) {
+        HueClient handler = getHueClient();
+        if (handler != null) {
+            FullGroup group = handler.getGroupById(groupId);
+            if (group != null) {
+                onGroupStateChanged(null, group);
+            }
+        }
```

Comment:
When this binding starts up this will get called for every channel. If you are just updating the channels with cached data it should be fine, but if you are making network requests or the like then you should use an expiring cache.

Reply:
It is just using the cache:
```
    public @Nullable FullGroup getGroupById(String groupId) {
        return lastGroupStates.get(groupId);
    }
```